picard/acousticbrainz/__init__.py

```diff
@@ -256,6 +256,13 @@ def ab_submit_features(tagger, file):
         submit_features_callback(file, None, None, True)
         return
 
+    # Set the filename. Especially on Windows the feature extracture fails to set the file basename only.
+    # As Picard already handles this just add the correct data ourselves.
+    try:
+        features['metadata']['tags']['file_name'] = file.base_filename
+    except KeyError:
+        log.warning('AcousticBrainz: Failed to set file_name in features data')
```

Comment:
Can you clarify in which case(s) this exception is triggered?

When it happens, does it still make sense to post data?

Reply:
It would if the structure returned by the extractor did not include metadata or tags. This really is just to handle this exception in case of invalid data.

I think if the data is still acceptable without this is something we can have the AB server decide. The thinking is that both streaming_extractor_music and AB server work together. So user might exchange streaming extractor music to a newer version, which maybe has different data. No need for the feature in Picard to break and crash just because of that.